src/viking.js

```diff
@@ -1,14 +1,73 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack() {
+    return this.strength;
+  }
+
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor (name, health, strength) {
+    super(health, strength);
+    this.name = name;
+    //this.vikStr = str;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    } else if (damage > this.health) {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry() {
+    return ("Odin Owns You All!");
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor (health, strength) {
+    super(health, strength);
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else if (damage > this.health) {
+      return `A Saxon has died in combat`;
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    receiveDamage(callback);
+    this.saxonArmy(pop);
+  }
```

Comment:
In the attack method, first thing you need to do is to get random element from the army array.
So take viking army array and using Math.random get one element (one army) from it.

After that,  take saxon army array and get one random element.

Then we need to implement the following task:
`should make a Saxon receiveDamage() equal to the strength of a Viking`

This means:
- randomly chosen saxon army gets damaged / call receiveDamage
- damage is equal to result of calling viking attack method